salt/_states/kubeconfig.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+from base64 import b64encode
+
+__virtualname__ = 'metalk8s_kubeconfig'
+
+
+def __virtual__():
+    return __virtualname__
+
+
+def managed(name, client_cert_info, config_context, **kwargs):
+    """Generate kubeconfig file with identities for control plane components"""
+    # TODO: manage idempotency
+    client_priv_key = __salt__['x509.create_private_key'](text=True)
+
+    client_cert = __salt__['x509.create_certificate'](
+        text=True,
+        public_key=client_priv_key,
+        **client_cert_info
+    )  # TODO; use ca server
+
+    ca_cert = client_cert_info.get('signing_cert')
+
+    with open(ca_cert, 'r') as fd:
```

Comment:
Can we use a `__salt__` function to achieve this?

Reply:
Ok, I will do. For the ca_cert, I'm thinking about storing a pem version of the ca cert in the pillar (base64), in order to spread it to all minion, so maybe I will drop this part, but still waiting for Teddy's PR for now to test

Reply:
Not in the pillar maybe in the mine it's better

Reply:
@ramanduh is this solved?